tests/test_bashcomplete.py

```diff
@@ -125,15 +125,16 @@ def asub(asub_opt):
 
     @cli.command('bsub')
     @click.option('--bsub-opt')
-    @click.argument('arg', type=click.Choice(['arg1', 'arg2']))
+    @click.argument('arg', type=click.Choice(['arg1', 'arg2']), required=True)
     def bsub(bsub_opt, arg):
         pass
 
     assert list(get_choices(cli, 'lol', [], '-')) == ['--cli-opt']
     assert list(get_choices(cli, 'lol', [], '')) == ['asub', 'bsub']
     assert list(get_choices(cli, 'lol', ['asub'], '-')) == ['--asub-opt']
     assert list(get_choices(cli, 'lol', ['asub'], '')) == ['bsub']
-    assert list(get_choices(cli, 'lol', ['bsub'], '')) == ['arg1', 'arg2', 'asub']
```

Comment:
Since this is a chained group, I don't see a reason why it would be zealous in any way to suggest another subcommand in this situation?

Reply:
@untitaker any thoughs on this?

Reply:
@untitaker ?

Reply:
I'm happy to merge this if you add a test that has optional arguments, for that case the other subcommands shouldn't be excluded.

Also please try to split up `get_choices`.

Reply:
Ok, I added a test case and simplified get_choices. After this goes in I also have a PR [here](https://github.com/pallets/click/pull/865) to add native ZSH completion support.